### ISSUE TITLE
renameManifestPackage does not change package name

### ISSUE THREAD

**Issue by pajah1** [iBotPeaches/Apktool#1753]
I'm not sure if I properly understand the purpose of the `renameManifestPackage` property in the `apktool.yml`, however if I understand correctly it should at least automatically change the `package` property in the `AndroidManifest.xml` file. However. this does not happen. Please, correct me if I misunderstood the purpose of the `renameManifestPackage` property.

### Information
1. Apktool Version - **2.3.1**
2. Operating System - **Windows 10**
3. **Any APK**

#### Decode:

```
I: Using Apktool 2.3.1 on vlc.apk
I: Loading resource table...
I: Decoding AndroidManifest.xml with resources...
S: WARNING: Could not write to (C:\Users\pajah\AppData\Local\apktool\framework), using C:\Users\pajah\AppData\Local\Temp\ instead...
S: Please be aware this is a volatile directory and frameworks could go missing, please utilize --frame-path if the default storage directory is unavailable
I: Loading resource table from file: C:\Users\pajah\AppData\Local\Temp\1.apk
I: Regular manifest package...
I: Decoding file-resources...
I: Decoding values */* XMLs...
I: Baksmaling classes.dex...
I: Copying assets and libs...
I: Copying unknown files...
I: Copying original files...
```

#### Build:

```
I: Using Apktool 2.3.1
I: Checking whether sources has changed...
I: Smaling smali folder into classes.dex...
I: Checking whether resources has changed...
I: Building resources...
S: WARNING: Could not write to (C:\Users\pajah\AppData\Local\apktool\framework), using C:\Users\pajah\AppData\Local\Temp\ instead...
S: Please be aware this is a volatile directory and frameworks could go missing, please utilize --frame-path if the default storage directory is unavailable
W: warning: string 'advanced_debugging' has no default translation.
W: warning: string 'beta_warning' has no default translation.
W: warning: string 'enable_brightness_gesture' has no default translation.
W: warning: string 'enable_iomx' has no default translation.
W: warning: string 'enable_jump_buttons' has no default translation.
W: warning: string 'enable_jump_buttons_summary' has no default translation.
W: warning: string 'enable_wheel_bar' has no default translation.
W: warning: string 'file_other' has no default translation.
W: warning: string 'filebrowser_title' has no default translation.
W: warning: string 'hardware_acceleration_error_message' has no default translation.
W: warning: string 'hardware_acceleration_error_title' has no default translation.
W: warning: string 'interface_secondary_display_category' has no default translation.
W: warning: string 'main_prefs_category' has no default translation.
W: warning: string 'new_app' has no default translation.
W: warning: string 'new_app_go' has no default translation.
W: warning: string 'new_app_msg' has no default translation.
W: warning: string 'play_as_audio_background_summary' has no default translation.
I: Copying libs... (/lib)
I: Building apk file...
I: Copying unknown files/dir...
```

### Steps to Reproduce
1. Download any APK (for example, [VLC](https://get.videolan.org/vlc-android/3.0.1/VLC-Android-3.0.1-ARMv7.apk))
2. `apktool d vlc.apk -o output/`
3. Change `renameManifestPackage` in `apktool.yml` to `com.hello.world`
4. `apktool b output/ -o vlc2.apk`
5. `apktool d vlc2.apk -o output2/`
6. `output2/AndroidManifest.xml` still contains the old package name. So what's the point of `renameManifestPackage`?

### APK
[VLC 3.0.1](https://get.videolan.org/vlc-android/3.0.1/VLC-Android-3.0.1-ARMv7.apk)

**Comment by ibuprophen1** [comment]
First off... your receiving "Warnings" and not "Errors". The warnings are just to warn you about something but, errors are to tell you about errors it had encountered that usually results in an apk that is bad, incomplete or nothing at all. 

I would state that changing the package name of an app via the Androidmanifest file works approx. 1/3 of the time/apps.

This is also based upon the complexity/dependency of the original package name. The more complex/dependent of the package name, the more work is involved.

There are apps that requires more than just changing the Androidmanifest but, also requires you to outline the new package name in one or more areas within the "res" & "smali" directories.

It's hard to explain because not all apps are the same for accomplishing this...

When it comes to the VLC... I believe that the VLC has a signature check when the app launches (this is for many of the more complex apps like YouTube).

Anyways good luck and, if needed, hopefully @iBotPeaches or someone else, can explain this better than I can.

I wish you the best of luck! :D

**Comment by pajah1** [comment]
@ibuprophen1 Thanks for your response. Warnings don't matter here at all. Also, I don't need the working APK on output (that's why I also don't need to change the `smali` structure and the old package name references), I'm just curious what does this property do. If I understand correctly, the non-null `renameManifestPackage` property calls the `aapt rename-manifest-package` command which *should* change `AndroidManifest.xml` `package` property.

**Comment by iBotPeaches** [comment]
`renameManifestPackage` is not for manual package renames with what you are attempting. At one point in the history of Android applications had advertised as one application at the manifest level, but defined a different package at the arsc level.

This led to a issues rebuilding, which we patched here: https://github.com/iBotPeaches/Apktool/commit/1e3b66f9be3e1c78afe20192e0c828e9520bcc2c

Applications needed a method (behind the scenes) to decode an application and rebuild with the correct package name despite showing a different one. It is not designed for manual package name changes. For the past part, most properties in the `apktool.yml` file are not changeable by the user as they are internal. You can of course change values like version/sdk-min in the yaml, but more complicated properties like renameManifest are not what you expect.

Perhaps with all the requests, a tutorial for renaming a package with apktool can be created. At this time, this isn't a bug though just misunderstanding of a property.

**Comment by weeebdev** [comment]
> `renameManifestPackage` is not for manual package renames with what you are attempting. At one point in the history of Android applications had advertised as one application at the manifest level, but defined a different package at the arsc level.
> 
> This led to a issues rebuilding, which we patched here: [1e3b66f](https://github.com/iBotPeaches/Apktool/commit/1e3b66f9be3e1c78afe20192e0c828e9520bcc2c)
> 
> Applications needed a method (behind the scenes) to decode an application and rebuild with the correct package name despite showing a different one. It is not designed for manual package name changes. For the past part, most properties in the `apktool.yml` file are not changeable by the user as they are internal. You can of course change values like version/sdk-min in the yaml, but more complicated properties like renameManifest are not what you expect.
> 
> Perhaps with all the requests, a tutorial for renaming a package with apktool can be created. At this time, this isn't a bug though just misunderstanding of a property.

Really need a tutorial for renaming a package. It still doesn't work

**Comment by anlexN** [comment]
@iBotPeaches @weeebdev 

Today, where is a tutorial for renaming a package? I really want to rename some apks' package name.
